web/modules/custom/dpl_admin/dpl_admin.install

```diff
@@ -32,3 +41,102 @@ function dpl_admin_requirements(string $phase): array {
+  $user_1_fields = array_diff_key($user_1->getFields(include_computed: FALSE), $excluded_fields);
+  $user_1_field_values = array_map(function (FieldItemListInterface $field) {
+    return $field->value;
+  }, $user_1_fields);
+
+  $result = \Drupal::database()->select('users_field_data')
+    ->fields('users_field_data', ['pass'])
+    ->condition('uid', $user_1->id())
+    ->execute();
+  if (!$result) {
+    throw new RuntimeException('Unable to retrieve password hash for user 1');
+  }
+  $password_hash = $result->fetchField();
+  $feedback[] = "Retrieved password hash for user 1";
+
+  // Transfer field data.
+  $new_user = (User::create($user_1_field_values))
+    ->addRole('local_administrator');
+
+  // Reset the existing user 1 account.
+  // Use random data for both username and password as this user is only meant
+  // to be used with onetime login links.
+  $random = new Random();
+  $user_1
+    ->set('field_author_name', "Administrator")
+    ->setUsername('admin_' . $random->name())
+    ->setEmail('')
+    ->setPassword($random->string(32));
+  $user_1->save();
+  $feedback[] = "Reset username and password for user 1";
+
+  $removed_roles = array_map(function (string $role) use ($user_1) {
+      $user_1->removeRole($role);
+      return $role;
+  }, $user_1->getRoles(exclude_locked_roles: TRUE));
+  $user_1->save();
+  $removed_roles_string = implode(', ', $removed_roles);
+  $feedback[] = "Removed roles {$removed_roles_string} for user 1";
+
+  $new_user->save();
+  $feedback[] = "Created new user {$new_user->id()} with local administrator role";
+
+  \Drupal::database()->update('users_field_data')
+    ->fields(['pass' => $password_hash])
+    ->condition('uid', $new_user->id())
+    ->execute();
+  $feedback[] = "Transferred password hash from user 1 to new user {$new_user->id()}";
+
+  // Migrate entity ownership to the new user.
+  $entities_by_types = array_map(function (EntityTypeInterface $entity_type) use ($user_1) {
+    try {
+      return \Drupal::entityTypeManager()->getStorage($entity_type->id())->loadByProperties(['uid' => $user_1->id()]);
+    }
+    catch (QueryException $e) {
+      // An exception if thrown for entity types without an uid property. Ignore
+      // these as they do not have relations to the existing user.
+      return [];
+    }
+  }, \Drupal::entityTypeManager()->getDefinitions());
+
+  /** @var array<string, array<EntityInterface&EntityOwnerInterface>> $user_owned_entities_by_types */
+  $user_owned_entities_by_types = array_filter($entities_by_types, function (array $entities) {
+    return count($entities) > 0 && reset($entities) instanceof EntityOwnerInterface;
+  });
+
+  $status_by_types = array_map(function (array $entities, string $entity_type) use ($new_user) {
+    $entities = array_map(fn (EntityOwnerInterface $entity) => $entity->setOwner($new_user), $entities);
+    $entities = array_map(fn (EntityInterface $entity) => $entity->save(), $entities);
```

Comment:
can you add a comment on what is happening here?

Reply:
I have tried to document the process by adding more explatory code comments.